scripts/utils.sh

```diff
@@ -31,3 +31,71 @@ function check_exit_code {
         fatal_error "${fail_msg}"
     fi
 }
+
+# Reimplement 'mkdir -p' with reporting on where permissions break down
```

Comment:
@ocaisa Maybe it makes sense to elaborate on the use case for this function a bit?
Why is `mkdir -p` not good enough, is it mainly to get better error reporting?

Can we add a reference to an issue on something here with more info?